### ISSUE TITLE
[1.4] Cutting in text boxes (Ctrl+X) sometimes crashes the game.

### ISSUE THREAD

**Issue by BlueStaggo** [tModLoader/tModLoader#1753]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Version

Latest public-1.4-alpha

### OS

Windows

### Platform

Steam

### Description

Sometimes, when you press Ctrl+X on a text box, tModLoader crashes. I keep accidentally crashing myself because I usually press Ctrl+X to clear the duplication menu search box.

### Log File

```
[11:18:52] [Main Thread/INFO] [tML]: Starting tModLoader client 1.4.2.3+0.12-1.4-Alpha-8305af16ddd7b2e2a959aa05831ab63bd3880daa-5249331498694505270
[11:18:52] [Main Thread/INFO] [tML]: Log date: 17/08/2021
[11:18:52] [Main Thread/INFO] [tML]: Running on Windows X64 NetCore 5.0.8
[11:18:52] [Main Thread/INFO] [tML]: Executable: C:\Program Files (x86)\Steam\steamapps\common\tModLoader\tModLoader.dll
[11:18:52] [Main Thread/INFO] [tML]: Working Directory: C:\Program Files (x86)\Steam\steamapps\common\tModLoader
[11:18:52] [Main Thread/INFO] [tML]: Developer mode enabled
[11:18:55] [Main Thread/DEBUG] [tML]: Native Resolve: FNA, Version=21.8.0.0, Culture=neutral, PublicKeyToken=null -> FNA3D
[11:18:55] [Main Thread/DEBUG] [tML]: 	attempting load C:\Program Files (x86)\Steam\steamapps\common\tModLoader\Libraries\Native\Windows\FNA3D.dll
[11:18:55] [Main Thread/DEBUG] [tML]: 	success
[11:18:56] [Main Thread/INFO] [tML]: Checking Steam installation...
[11:18:56] [Main Thread/DEBUG] [tML]: Native Resolve: Steamworks.NET, Version=15.0.1.0, Culture=neutral, PublicKeyToken=null -> steam_api64
[11:18:56] [Main Thread/DEBUG] [tML]: 	attempting load C:\Program Files (x86)\Steam\steamapps\common\tModLoader\Libraries\Native\Windows\steam_api64.dll
[11:18:56] [Main Thread/DEBUG] [tML]: 	success
[11:18:56] [Main Thread/INFO] [tML]: Steam installation OK.
[11:18:56] [Main Thread/INFO] [Terraria]: Found Terraria steamapp install at: C:\Program Files (x86)\Steam\steamapps\common\Terraria
[11:18:56] [Main Thread/INFO] [Terraria]: Steam Cloud Quota: 100.3 MB available
[11:18:57] [Main Thread/INFO] [FNA]: FNA3D Driver: D3D11
[11:18:57] [Main Thread/INFO] [FNA]: D3D11 Adapter: AMD Radeon(TM) R5 Graphics
[11:18:58] [Main Thread/DEBUG] [Terraria]: Device Created, Adapter: Generic PnP Monitor, DisplayMode: {{Width:1920 Height:1080 Format:Color}}, Profile: Reach, Width: 800, Height: 480, Fullscreen: False, Display: \\.\DISPLAY1
[11:18:58] [Main Thread/DEBUG] [Terraria]: Device Reset, Adapter: Generic PnP Monitor, DisplayMode: {{Width:1920 Height:1080 Format:Color}}, Profile: Reach, Width: 800 -> 1920, Height: 480 -> 1080, Fullscreen: False, Display: \\.\DISPLAY1
[11:18:59] [Main Thread/DEBUG] [tML]: Native Resolve: FNA, Version=21.8.0.0, Culture=neutral, PublicKeyToken=null -> FAudio
[11:18:59] [Main Thread/DEBUG] [tML]: 	attempting load C:\Program Files (x86)\Steam\steamapps\common\tModLoader\Libraries\Native\Windows\FAudio.dll
[11:18:59] [Main Thread/DEBUG] [tML]: 	success
[11:19:45] [33/INFO] [tML]: Unloading mods
[11:19:47] [33/INFO] [tML]: Finding Mods...
[11:19:47] [33/INFO] [tML]: Sandboxing: CheatSheet
[11:19:47] [33/INFO] [tML]: Sandboxing: GenericClass
[11:19:47] [33/INFO] [tML]: Sandboxing: StaggosGoodVisuals
[11:19:47] [33/INFO] [tML]: Instantiating Mods...
[11:19:48] [33/INFO] [tML]: Initializing: ModLoader v0.12
[11:19:49] [33/INFO] [tML]: Initializing: CheatSheet v0.7.0.1
[11:19:49] [33/INFO] [tML]: Initializing: GenericClass v0.1
[11:19:49] [33/INFO] [tML]: Initializing: StaggosGoodVisuals v0.2
[11:19:50] [33/INFO] [tML]: Setting up...
[11:19:50] [33/INFO] [tML]: Loading: ModLoader v0.12
[11:19:50] [33/INFO] [tML]: Loading: CheatSheet v0.7.0.1
[11:19:51] [33/INFO] [tML]: Loading: GenericClass v0.1
[11:19:51] [33/INFO] [tML]: Loading: StaggosGoodVisuals v0.2
[11:19:51] [33/INFO] [tML]: RAM usage: 1.3 GB
[11:19:52] [33/INFO] [tML]: Adding Recipes...
[11:22:35] [32/INFO] [StatusText]: Resetting game objects
[11:22:37] [32/INFO] [StatusText]: Loading world data
[11:22:38] [32/INFO] [StatusText]: Settling liquids
[11:22:42] [32/INFO] [StatusText]: Loading map data
[11:22:42] [32/INFO] [StatusText]: Drawing map
[11:22:43] [Main Thread/INFO] [Terraria]: Entering world with player: BlueStag, IsCloud=False
```

### Steps to reproduce

1. Open up the Journey Power menu.
2. Open up the duplication menu.
3. Type something in.
4. Press Ctrl+X to cut.
5. Repeat steps 3 & 4 until the game crashes.

### Expected Behavior

The text is cut (the text has been removed but it is in my clipboard) without the game crashing.

### Actual Behavior

The game crashes.

### Reproduction frequency

This issue is not too common.

### Additional Information

No text should be selected to reproduce this issue.

**Comment by Steviegt6** [comment]
Likely an issue with the platform's `IWindowService` or whatever interface takes care of clipboard stuff in `ReLogic`. I'd assume it's directly relating to the pasting issues we have as well.

**Comment by Chicken-Bones** [comment]
Fixed in #1757
